### ISSUE TITLE
Expectation that solvers don't need access to the problem until `solve()`

### ISSUE THREAD

**Issue by zsunberg** [JuliaPOMDP/POMDPs.jl#80]
Hi all, I can't recall if we already made a decision on this. If we have, please remind me.

I like the elegance of the solver constructor being completely independent of the problem e.g. the following should work fine:

```julia
solver = MySolver(#= nothing that is problem dependent =#)

problem1 = BabyPOMDP()
problem2 = TigerPOMDP()

policy1 = solve(solver, problem1)
policy2 = solve(solver, problem2)
```

Should this be an expectation of solvers? What do you all think? I think it is technically possible, but might require some extra lines of code in some cases.

**Comment by mykelk** [comment]
Yes, this should be the expectation.

**Comment by etotheipluspi** [comment]
I'm ok with this as well. 

I briefly looked through our solvers, and it looks like DESPOT is the only solver that takes a POMDP type as an input to the constructor. @ebalaban will this be a problem, or can you work around it.

**Comment by ebalaban** [comment]
Apologies for the delayed response on this. I removed the dependency on the POMDP. I will try to remove the dependency on the initial belief a bit later today. In practice, what @zsunberg envisions in the first post will still only be possible if the `problem1` and `problem2` are modeled using the same data types, as the solver is now heavily parametrized for speed.

**Comment by ebalaban** [comment]
Fixed with 319d35b. I'll close this issue then, but if someone wants to do anything more on it, please reopen.
